dialogflow/context_management.rb

```diff
@@ -0,0 +1,116 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require 'securerandom'
+
+def list_contexts project_id:, session_id:
+  # [START dialogflow_list_contexts]
+  # project_id = "Your Google Cloud project ID"
+  # session_id = "existing_session_id"
+  
+  require "google/cloud/dialogflow"
+
+  contexts_client = Google::Cloud::Dialogflow::Contexts.new
+  session_path = contexts_client.class.session_path project_id, session_id
+
+  contexts = contexts_client.list_contexts(session_path)
+
+  puts "Contexts for session #{session_path}:\n"
+  contexts.each do |context|
+    puts "Context name:   #{context.name}"
+    puts "Lifespan count: #{context.lifespan_count}"
+    if !context.parameters.nil?
```

Comment:
We have falsey/truthy conditionals (non-nil is truthy)

```ruby
if context.parameters
  # some stuff
end
```

That said, if you want the sample to read more like English/prose, it's not uncommon to add explicit checks using predicate methods to make the sample more clear:

```ruby
if context.parameters.any? # <-- recommended :)
  # some stuff
end
```

Reply:
done